### PR TITLE
Post training upload prediction labels

### PR DESCRIPTION
**Pull request by samiyaakhtar**
- New API to get classification mapping for names passed as query param
- POST api in /labels to post prediction labels after training

### CODE REVIEW

File: functions/pipeline/shared/db_access/db_access_v2.py [hunk 1]
```diff
@@ -316,6 +316,31 @@ def checkout_images(self, image_count, user_id):
             conn.close()
         return list(image_id_to_image_labels.values())
 
+    def get_classification_mapping(self, classification_names):
```
Comment (andrebriggs) [inline functions/pipeline/shared/db_access/db_access_v2.py]: @samiyaakhtar We already have get_classification_map(), Maybe use or modify that so we don't have two methods for the same thing.

File: train/training.py [hunk 2]
```diff
@@ -93,6 +94,61 @@ def convert_labels_to_csv(data, tagging_output_file_path):
...
+def upload_data_post_training(training_id):
+    tagging_output_file_path =  os.path.expanduser(config.get('tagged_output'))
+    function_url = config.get("url") + "/api/classification"
+    query = {
+        "userName": config.get('tagging_user')
+    }
+
+    # First, we need to get a mapping of class names to class ids
+    classes = ""
+    with open(tagging_output_file_path + "/tagged.csv") as f:
```
Comment (andrebriggs) [inline train/training.py]: @samiyaakhtar we should pass in the entire path as a param. No hardcoding of file names

File: train/training.py [hunk 3]
```diff
@@ -93,6 +94,61 @@ def convert_labels_to_csv(data, tagging_output_file_path):
...
+    response = requests.get(function_url, params=query)
+    classification_name_to_class_id = response.json()
+    
+    # Now that we have a mapping, we create prediction labels in db
+    query = {
+        "userName": config.get('tagging_user'),
+        "trainingId": training_id
+    }
+    function_url = config.get("url") + "/api/labels"
+    payload_json = process_post_training_csv(tagging_output_file_path + "/tagged.csv", training_id, classification_name_to_class_id)
```
Comment (andrebriggs) [inline train/training.py]: @samiyaakhtar the post training file (prediction) is actually not this file. Let's have a method parameter called prediction_output_file or something. 

``` 
def upload_data_post_training(training_id, prediction_output_file):
...
```

If possible it's pass all arguments in the method signature. That way it becomes a static method and easier to test

File: train/training.py [hunk 4]
```diff
@@ -93,6 +94,61 @@ def convert_labels_to_csv(data, tagging_output_file_path):
...
+    requests.post(function_url, params=query, json=payload_json)
+
+def process_post_training_csv(csv_path, training_id, classification_name_to_class_id):
+    payload_json = []
+    with open(csv_path) as f:
+        content = f.readlines()
+        count = 0
+        for line in content:
+            if count > 0:
```
Comment (andrebriggs) [inline train/training.py]: @samiyaakhtar I assume we are doing this to skip the head row. Also we are manually splitting the lines by comma. An approach with less code could be:

```
import csv
with open(csv_path) as f:
         reader = csv.reader(f)
         next(reader, None) #Skip header
         for row in reader:
               ....
               # Use row[n] to access columns 
```

**Review by andrebriggs — CHANGES_REQUESTED** [review]

File: train/training.py (hunk 2 above)
Comment (samiyaakhtar) [reply train/training.py]: Done

File: train/training.py (hunk 3 above)
Comment (samiyaakhtar) [reply train/training.py]: Done

File: train/training.py (hunk 4 above)
Comment (samiyaakhtar) [reply train/training.py]: Done

File: functions/pipeline/shared/db_access/db_access_v2.py [hunk 5]
```diff
@@ -1,97 +1,11 @@
...
-                image_height, image_width, box_confidence=0, image_confidence= 0):
-        super().__init__(image_id, classification_id, x_min, x_max, y_min, y_max)
-        self.training_id = training_id
-        self.image_height = image_height
-        self.image_width = image_width
-        self.box_confidence = box_confidence
-        self.image_confidence = image_confidence
-
-
+from .models import ImageTag, ImageLabel, ImageTagState, AnnotatedLabel, Tag, ImageInfo, PredictionLabel
```
Comment (andrebriggs) [inline functions/pipeline/shared/db_access/db_access_v2.py:8]: @samiyaakhtar Does it work if we use absolute references?

**Review by andrebriggs — APPROVED** [review]

File: functions/pipeline/shared/db_access/db_access_v2.py (hunk 5 above)
Comment (samiyaakhtar) [reply functions/pipeline/shared/db_access/db_access_v2.py:8]: It doesn't work with absolute since functions are deployed from inside functions/pipeline so it has to be within this folder. It can be used outside with functions.pipeline.shared.db_access..
